annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,322 @@
+    (
+        (
+            {
+                "file_id": 1,
+                "job_id": 2,
+                "pages": [1, 2, 4],
+            },
+            Mock(spec=File, pages_number=3),
+            r"pages \(\{4\}\) do not belong to file",
+        ),
+        (
+            {
+                "file_id": 1,
+                "job_id": 2,
+                "pages": [1, 2, 3],
+            },
+            None,
+            r"file with id 1 is not assigned for job 2",
+        ),
+    ),
+)
+def test_validate_files_info_raises_error_for_invalid_file_info(
+    mock_session: Mock,
+    task_info: Dict[str, Union[int, List[int]]],
+    mock_file: Mock,
+    expected_error_message_regex: str,
+):
+    db_session = mock_session()
+    db_session.query().filter_by().first.return_value = mock_file
+
+    with pytest.raises(
+        FieldConstraintError, match=expected_error_message_regex
+    ):
+        validate_files_info(db_session, task_info)
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+def test_check_cross_annotating_pages(mock_session: Mock):
+    db_session = mock_session()
+    task_info = {"user_id": 1, "file_id": 2, "job_id": 3, "pages": {4, 5}}
+    existing_pages = []
+
+    mock_query = db_session.query.return_value
+    mock_query.filter.return_value.all.return_value = [(existing_pages,)]
+
+    try:
+        check_cross_annotating_pages(db_session, task_info)
+
+        assert db_session.query.call_count == 1
+        db_session.query.assert_has_calls(
+            [
+                call(ManualAnnotationTask.pages),
+            ]
+        )
+        mock_query.filter.assert_called_once()
+
+    except FieldConstraintError:
+        pytest.fail("FieldConstraintError was raised unexpectedly")
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+def test_check_cross_annotating_pages_raise_error_when_page_already_annotated(
+    mock_session: Mock,
+):
+    db_session = mock_session()
+    task_info = {"user_id": 1, "file_id": 2, "job_id": 3, "pages": {4, 5}}
+    existing_pages = [4, 5]
+
+    mock_query = db_session.query.return_value
+    mock_query.filter.return_value.all.return_value = [(existing_pages,)]
+
+    expected_error_message = (
+        "within cross validation job user can't validate file's pages "
+        "that are already distributed in annotation tasks for this user: "
+        "{4, 5}"
+    )
+
+    with pytest.raises(FieldConstraintError, match=expected_error_message):
+        check_cross_annotating_pages(db_session, task_info)
+
+
+@pytest.mark.parametrize(
+    (
+        "is_validation",
+        "failed",
+        "annotated",
+        "not_processed",
+        "annotation_user",
+        "validation_user",
+        "expected_error_message",
+    ),
+    (
+        (
+            True,
+            {1},
+            set(),
+            set(),
+            False,
+            True,
+            "Missing `annotation_user_for_failed_pages` param "
+            "for failed pages.",
+        ),
```

Comment:
For this and other test cases: 
1. is_validation is always true - remove it
2. no need to check for the whole error message - you can match important parts which are different from each other. And simply looking at this I would split this test into multiple test cases (but only if they are logically separable)

Reply:
Ok.